### ISSUE TITLE
docker-compose, when a large amount of logs are output from the container, it cannot be confirmed by the tail command on the host side.

### ISSUE THREAD

**Issue by katoharuki** [docker/compose#9516]
**Description**

I am building docker-compose with macos as the host.

When a large amount of logs are output in a docker container that shares a volume with the host  
I can tail from the container without any problems, but I cannot tail from the host.  
After restarting the container, I can see the log from the host as well.

Describes the reproduction procedure and docker version information.

**Steps to reproduce the issue:**
- compose
```
$ cat docker-compose.yml 
---
version: "3.7"

services:
  web:
    image: centos
    volumes:
      - ./not-mutagen-sync-volume:/not-mutagen-container-volume
    tty: true
```
- Log from the container
```
[root@f832f0c09889 /]# nohup bash -c 'i=0 ;while true; do echo $(date "+%Y%m%d-%H%M%S") $(( i++ ))  && cat /dev/urandom | base64 >> /not-mutagen-container-volume/tmp ;sleep 0.01; done' &
[root@f832f0c09889 /]#
[root@f832f0c09889 /]#
[root@f832f0c09889 /]# tail not-mutagen-container-volume/tmp 
38M2DRXGuUMUV5fRLBRbP4FmDKZcZwoF4Ifo5YNERIEUvDz4pMXqljUs1wLG9i6vRxXFWk4jtUJT
:
```
- Not visible to the host
```
$ tail not-mutagen-sync-volume/tmp 

^C
```
- Container restart
```
$ docker-compose stop web
Stopping mutagen-test_web_1 ... done
%                                                                                                                                                         
$ docker-compose start web
Starting web ... done
```
- I could see the log from the host as well.
```
$ tail not-mutagen-sync-volume/tmp
38M2DRXGuUMUV5fRLBRbP4FmDKZcZwoF4Ifo5YNERIEUvDz4pMXqljUs1wLG9i6vRxXFWk4jtUJT
:
```

**Describe the results you received:**

When the log is output by the container, it is not updated in real time on the host, and the log on the host side is not updated unless the container is restarted.

**Describe the results you expected:**

After outputting the log in the container, I can check it in real time on the host with the tail command.

**Additional information you deem important (e.g. issue happens only occasionally):**

It does not occur when the amount of logs is small.
In this case, if the character string output by the base command is one line at the same time, it can be confirmed without any problem.

**Output of `docker compose version`:**
**Output of `docker info`:**
**Additional environment details:**

```
The host is MacOS.
$ sw_vers
ProductName:    macOS
ProductVersion: 11.6.2
BuildVersion:   20G314

The docker compose version.
$ docker-compose version
docker-compose version 1.29.1, build c34c88b2
docker-py version: 5.0.0
CPython version: 3.9.0
OpenSSL version: OpenSSL 1.1.1h  22 Sep 2020

The output of docker version.
$ docker --version
Docker version 20.10.6, build 370c289

The output of docker context show.
$ docker context show 
default
$ docker context inspect default
[
    {
        "Name": "default",
        "Metadata": {
            "StackOrchestrator": "swarm"
        },
        "Endpoints": {
            "docker": {
                "Host": "unix:///var/run/docker.sock",
                "SkipTLSVerify": false
            },
            "kubernetes": {
                "Host": "https://xxxx.ap-northeast-1.eks.amazonaws.com",
                "SkipTLSVerify": false,
                "DefaultNamespace": "default",
                "Exec": {
                    "command": "aws-iam-authenticator",
                    "args": [
                        "token",
                        "-i",
                        "xxx"
                    ],
                    "env": [
                        {
                            "name": "AWS_STS_REGIONAL_ENDPOINTS",
                            "value": "regional"
                        },
                        {
                            "name": "AWS_DEFAULT_REGION",
                            "value": "ap-northeast-1"
                        },
                        {
                            "name": "AWS_PROFILE",
                            "value": "xxx"
                        }
                    ],
                    "apiVersion": "client.authentication.k8s.io/v1beta1"
                }
            }
        },
        "TLSMaterial": {
            "kubernetes": [
                "ca.pem"
            ]
        },
        "Storage": {
            "MetadataPath": "\u003cIN MEMORY\u003e",
            "TLSPath": "\u003cIN MEMORY\u003e"
        }
    }
]

The output of docker info.
$ docker info 
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)
  compose: Docker Compose (Docker Inc., 2.0.0-beta.1)
  scan: Docker Scan (Docker Inc., v0.8.0)

Server:
 Containers: 31
  Running: 16
  Paused: 0
  Stopped: 15
 Images: 32
 Server Version: 20.10.6
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc version: 12644e614e25b05da6fd08a38ffa0cfe1903fdec
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.10.25-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 4
 Total Memory: 1.942GiB
 Name: docker-desktop
 ID: xxx
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Comment by ndeloof** [comment]
Sorry for coming late on this one
If still relevant, could you please check this is a compose or a generic docker (desktop?) issue by running the same container with a direct `docker run -it  -v $(pwd)/not-mutagen-sync-volume:/not-mutagen-container-volume centos <command>`

**Comment by katoharuki** [comment]
Sorry for the late reply.

The problem was resolved when I upgraded Docker Desktop from 4.2.0 to 4.9.0.

I think it's a Docker Desktop issue.

Thank you for your response.
